Core_Pregnancy/PregnancyPlugin.Hooks.cs

```diff
@@ -0,0 +1,269 @@
+                    if (heroine == null) throw new ArgumentNullException(nameof(heroine));
+
+                    var results = new List<ChaFileControl>();
+
+                    if (heroine.ChaControl != null && heroine.ChaControl.chaFile != null)
+                        results.Add(heroine.ChaControl.chaFile);
+
+                    var npc = heroine.GetNPC();
+                    if (npc != null && npc.ChaControl != null && npc.ChaControl.chaFile != null)
+                        results.Add(npc.ChaControl.chaFile);
+
+                    return results;
+                }
+
+                private static bool CanGetSpawned(AgentActor heroine)
+                {
+                    var isOnLeave = GetRelatedChaFiles(heroine)
+                        .Any(c =>
+                        {
+                            var pd = PregnancyData.Load(ExtendedSave.GetExtendedDataById(heroine.ChaControl.chaFile, GUID));
+                            if (pd == null) return false;
+                            return pd.GameplayEnabled && pd.Week >= PregnancyData.LeaveSchoolWeek;
+                        });
+                    return !isOnLeave;
+                }
+
+                private static List<AgentActor> GetFilteredHeroines(List<AgentActor> originalList)
+                {
+                    var filteredHeroines = originalList.Where(CanGetSpawned).ToList();
+                    return filteredHeroines;
+                }       
+
+                #region InflationAI                
+
+
+
+                [HarmonyPrefix]
+                [HarmonyPatch(typeof(Sonyu), "Proc", typeof(int), typeof(HScene.AnimationListInfo))]
+                public static void Sonyu_Proc(Sonyu __instance)
+                {                    
+                    //Get current user button click type
+                    var ctrlFlag = Traverse.Create(__instance).Field("ctrlFlag").GetValue<HSceneFlagCtrl>();                                    
+                    DetermineInflationState(ctrlFlag);                   
+                }
+
+
+                [HarmonyPrefix]
+                [HarmonyPatch(typeof(Houshi), "Proc", typeof(int), typeof(HScene.AnimationListInfo))]
+                public static void Houshi_Proc(Houshi __instance)
+                {                    
+                    //Get current user button click type
+                    var ctrlFlag = Traverse.Create(__instance).Field("ctrlFlag").GetValue<HSceneFlagCtrl>(); 
+                    DetermineInflationState(ctrlFlag);                                   
+                }
+
+
+                //When user clicks finish button, set the inflation based on the button clicked
+                private static void DetermineInflationState(HSceneFlagCtrl ctrlFlag)
+                {
+                    //swallow
+                    if (ctrlFlag.click == HSceneFlagCtrl.ClickKind.FinishInSide 
+                        || ctrlFlag.click == HSceneFlagCtrl.ClickKind.FinishSame  
```

Comment:
HSceneFlagCtrl appears to be the equivalent to HFlag in KK.  Unfortunately its not quite as straight forward to hook.